lib/kitchen/driver/azurerm.rb

```diff
@@ -224,8 +224,20 @@ def create(state)
           vaultResourceGroup: config[:vault_resource_group],
         }
 
+        # Set a randomly generated password or one from the environment if an SSH key isn't present
         if instance.transport[:ssh_key].nil?
+
+          # Set our generated password
           deployment_parameters["adminPassword"] = config[:password]
+
+          # Override the generated password with the environment if set
+          if not ENV['KITCHEN_VM_PASSWORD'].nil? 
```

Comment:
I'm not sure I understand what this change has to do with #118. My usage of `ENV['KITCHEN_VM_PASSWORD']` was just to illustrate the state persistence issue in an easily reproducible way, without having to edit the `.kitchen.yml`. It could also be reproduced by creating a `.kitchen.yml` with an empty `password:` value, running `kitchen create` and seeing that it fails, then populating the `password:` value in the `.kitchen.yml` with something valid and then running `kitchen create` again and observing that it still fails despite the `.kitchen.yml` now being valid.

Supporting the use of a `KITCHEN_VM_PASSWORD` environment variable doesn't seem necessary since this can already be done via the `password:` driver property in the `.kitchen.yml`, and if the user wishes to populate that with the value of an environment variable, they can already do that as shown in my repro, so this doesn't really add any functionality that doesn't already exist, aside from being irrelevant to the issue.